plugins/amazonq/chat/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/amazonq/workspace/context/EncoderServerTest.kt

```diff
@@ -0,0 +1,47 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.amazonq.workspace.context
+
+import com.intellij.openapi.project.Project
+import com.intellij.util.io.HttpRequests
+import org.apache.commons.codec.digest.DigestUtils
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import software.aws.toolkits.jetbrains.services.cwc.editor.context.project.EncoderServer
+import software.aws.toolkits.jetbrains.utils.rules.CodeInsightTestFixtureRule
+import software.aws.toolkits.jetbrains.utils.rules.JavaCodeInsightTestFixtureRule
+
+class EncoderServerTest {
+    @Rule @JvmField
+    val projectRule: CodeInsightTestFixtureRule = JavaCodeInsightTestFixtureRule()
+    private val testUrl = "https://amazon.com"
+    internal lateinit var project: Project
+    private lateinit var encoderServer: EncoderServer
+
+    @Before
+    open fun setup() {
+        project = projectRule.project
+        encoderServer = EncoderServer(project)
+    }
+
+    @Test
+    fun `test download artifacts validate hash if it does not match`() {
+        val inputBytes = HttpRequests.request(testUrl).readBytes(null)
```

Comment:
just generate some random bytes; we don't need to use the internet here

Reply:
alternatively, change to test against the real thing and move to the `it/` folder tree so it doesn't run as part of unit tests